### ISSUE TITLE
remove_duplicates.sh aborts pipeline

### ISSUE THREAD

**Issue by iromeo** [JetBrains-Research/washu#20]
Current `remove_duplicates.sh` has some log4j misconfiguration that's why it finishes successfully but writes related error to log file and check_logs aborts pipeline.

I think we need to fix `remove_duplicates.sh` in order not to output any errors

Error:
```
bash /home/rcherniatchik/work/washu/parallel/remove_duplicates.sh ~/picard.jar /scratch/rcherniatchik/geo/GSM1003475_bams /scratch/rcherniatchik/geo/GSM1003508_bams /scratch/rcherniatchik/geo/GSM1057012_bams /scratch/rcherniatchik/geo/GSM1057015_bams /scratch/rcherniatchik/geo/GSM1057026_bams /scratch/rcherniatchik/geo/GSM1057027_bams /scratch/rcherniatchik/geo/GSM2695650_bams /scratch/rcherniatchik/geo/GSM2695651_bams /scratch/rcherniatchik/geo/GSM2695652_bams /scratch/rcherniatchik/geo/GSM2695653_bams
Traceback (most recent call last):
  File "pipeline_tf.py", line 260, in <module>
    cli()
  File "pipeline_tf.py", line 30, in cli
    run_pipeline(args.out, args.data)
  File "pipeline_tf.py", line 157, in run_pipeline
    lambda dirs: run_bash("parallel/remove_duplicates.sh",
  File "pipeline_tf.py", line 250, in process_dirs
    processor_fun(list(zip(*dirs_to_process))[0])
  File "pipeline_tf.py", line 158, in <lambda>
    picard_tools, *dirs))
  File "/home/rcherniatchik/work/washu/pipeline_utils.py", line 37, in run_bash
    subprocess.run(command, shell=True, check=True)
  File "/home/rcherniatchik/miniconda3/envs/py35/lib/python3.5/subprocess.py", line 398, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'bash /home/rcherniatchik/work/washu/parallel/remove_duplicates.sh ~/picard.jar /scratch/rcherniatchik/geo/GSM1003475_bams /scratch/rcherniatchik/geo/GSM1003508_bams /scratch/rcherniatchik/geo/GSM1057012_bams /scratch/rcherniatchik/geo/GSM1057015_bams /scratch/rcherniatchik/geo/GSM1057026_bams /scratch/rcherniatchik/geo/GSM1057027_bams /scratch/rcherniatchik/geo/GSM2695650_bams /scratch/rcherniatchik/geo/GSM2695651_bams /scratch/rcherniatchik/geo/GSM2695652_bams /scratch/rcherniatchik/geo/GSM2695653_bams' returned non-zero exit status 1
```

`# GSM1003475_hg19_unique.log`
```
--------------------------------------------------------------
Begin PBS Prologue Fri Oct  6 15:45:36 CDT 2017 1507322736
Job ID:                 3512570.mgt2.cluster
Username:               rcherniatchik
Group:                  martyomov
Nodes:                  gpu14
End PBS Prologue Fri Oct  6 15:45:36 CDT 2017 1507322736
--------------------------------------------------------------
Picked up _JAVA_OPTIONS: -Xmx1g
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console.
[Fri Oct 06 15:45:38 CDT 2017] picard.sam.markduplicates.MarkDuplicates INPUT=[GSM1003475_hg19.bam] OUTPUT=GSM1003475_hg19_unique.bam METRICS_FILE=GSM1003475_hg19_metrics.txt REMOVE_DUPLICATES=true    MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 TAG_DUPLICATE_SET_MEMBERS=false REMOVE_SEQUENCING_DUPLICATES=false TAGGING_POLICY=DontTag ASSUME_SORTED=false DUPLICATE_SCORING_STRATEGY=SUM_OF_BASE_QUALITIES PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates READ_NAME_REGEX=<optimized capture of last three ':' separated fields as numeric values> OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false
[Fri Oct 06 15:45:38 CDT 2017] Executing as rcherniatchik@gpu14 on Linux 2.6.32-642.11.1.el6.centos.plus.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_40-b26; Deflater: Intel; Inflater: Intel; Picard version: 2.10.5-SNAPSHOT
INFO    2017-10-06 15:45:38     MarkDuplicates  Start of doWork freeMemory: 1015680464; totalMemory: 1029177344; maxMemory: 1029177344
INFO    2017-10-06 15:45:38     MarkDuplicates  Reading input file and constructing read end information.
INFO    2017-10-06 15:45:38     MarkDuplicates  Will retain up to 3728903 data points before spilling to disk.
WARNING 2017-10-06 15:45:38     AbstractOpticalDuplicateFinderCommandLineProgram        A field field parsed out of a read name was expected to contain an integer and did not. Read name: SRR568246.22000042. Cause: String 'SRR568246.22000042' did not start with a parsable number.
INFO    2017-10-06 15:45:43     MarkDuplicates  Read     1,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    4s.  Last read position: chr8:66,365,819
INFO    2017-10-06 15:45:43     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:45:46     MarkDuplicates  Read     2,000,000 records.  Elapsed time: 00:00:07s.  Time for last 1,000,000:    3s.  Last read position: chr8:130,219,254
INFO    2017-10-06 15:45:46     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:45:49     MarkDuplicates  Read     3,000,000 records.  Elapsed time: 00:00:11s.  Time for last 1,000,000:    3s.  Last read position: chr18:52,531,041
INFO    2017-10-06 15:45:49     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:45:54     MarkDuplicates  Read     4,000,000 records.  Elapsed time: 00:00:15s.  Time for last 1,000,000:    4s.  Last read position: chr15:26,790,461
INFO    2017-10-06 15:45:54     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:45:58     MarkDuplicates  Read     5,000,000 records.  Elapsed time: 00:00:19s.  Time for last 1,000,000:    3s.  Last read position: chr15:95,196,796
INFO    2017-10-06 15:45:58     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:01     MarkDuplicates  Read     6,000,000 records.  Elapsed time: 00:00:22s.  Time for last 1,000,000:    3s.  Last read position: chr10:62,534,868
INFO    2017-10-06 15:46:01     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:04     MarkDuplicates  Read     7,000,000 records.  Elapsed time: 00:00:25s.  Time for last 1,000,000:    3s.  Last read position: chr10:128,847,887
INFO    2017-10-06 15:46:04     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:09     MarkDuplicates  Read     8,000,000 records.  Elapsed time: 00:00:30s.  Time for last 1,000,000:    4s.  Last read position: chrX:68,770,809
INFO    2017-10-06 15:46:09     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:12     MarkDuplicates  Read     9,000,000 records.  Elapsed time: 00:00:33s.  Time for last 1,000,000:    3s.  Last read position: chrX:141,389,481
INFO    2017-10-06 15:46:12     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:17     MarkDuplicates  Read    10,000,000 records.  Elapsed time: 00:00:38s.  Time for last 1,000,000:    4s.  Last read position: chr13:72,615,417
INFO    2017-10-06 15:46:17     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:20     MarkDuplicates  Read    11,000,000 records.  Elapsed time: 00:00:41s.  Time for last 1,000,000:    3s.  Last read position: chr19:29,583,756
INFO    2017-10-06 15:46:20     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:25     MarkDuplicates  Read    12,000,000 records.  Elapsed time: 00:00:46s.  Time for last 1,000,000:    4s.  Last read position: chr6:45,151,109
INFO    2017-10-06 15:46:25     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:28     MarkDuplicates  Read    13,000,000 records.  Elapsed time: 00:00:49s.  Time for last 1,000,000:    3s.  Last read position: chr6:112,405,908
INFO    2017-10-06 15:46:28     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:31     MarkDuplicates  Read    14,000,000 records.  Elapsed time: 00:00:52s.  Time for last 1,000,000:    3s.  Last read position: chr12:5,509,251
INFO    2017-10-06 15:46:31     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:37     MarkDuplicates  Read    15,000,000 records.  Elapsed time: 00:00:58s.  Time for last 1,000,000:    5s.  Last read position: chr12:74,533,876
INFO    2017-10-06 15:46:37     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:40     MarkDuplicates  Read    16,000,000 records.  Elapsed time: 00:01:01s.  Time for last 1,000,000:    3s.  Last read position: chr16:8,968,436
INFO    2017-10-06 15:46:40     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:43     MarkDuplicates  Read    17,000,000 records.  Elapsed time: 00:01:05s.  Time for last 1,000,000:    3s.  Last read position: chr14:20,728,465
INFO    2017-10-06 15:46:43     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:47     MarkDuplicates  Read    18,000,000 records.  Elapsed time: 00:01:08s.  Time for last 1,000,000:    3s.  Last read position: chr14:86,600,222
INFO    2017-10-06 15:46:47     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:51     MarkDuplicates  Read    19,000,000 records.  Elapsed time: 00:01:13s.  Time for last 1,000,000:    4s.  Last read position: chr3:44,691,306
INFO    2017-10-06 15:46:51     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:55     MarkDuplicates  Read    20,000,000 records.  Elapsed time: 00:01:16s.  Time for last 1,000,000:    3s.  Last read position: chr3:113,017,336
INFO    2017-10-06 15:46:55     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:46:59     MarkDuplicates  Read    21,000,000 records.  Elapsed time: 00:01:20s.  Time for last 1,000,000:    3s.  Last read position: chr3:177,461,001
INFO    2017-10-06 15:46:59     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:03     MarkDuplicates  Read    22,000,000 records.  Elapsed time: 00:01:24s.  Time for last 1,000,000:    3s.  Last read position: chr9:76,325,665
INFO    2017-10-06 15:47:03     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:07     MarkDuplicates  Read    23,000,000 records.  Elapsed time: 00:01:28s.  Time for last 1,000,000:    4s.  Last read position: chr4:2,801,890
INFO    2017-10-06 15:47:07     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:10     MarkDuplicates  Read    24,000,000 records.  Elapsed time: 00:01:32s.  Time for last 1,000,000:    3s.  Last read position: chr4:72,465,525
INFO    2017-10-06 15:47:10     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:14     MarkDuplicates  Read    25,000,000 records.  Elapsed time: 00:01:35s.  Time for last 1,000,000:    3s.  Last read position: chr4:137,791,361
INFO    2017-10-06 15:47:14     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:17     MarkDuplicates  Read    26,000,000 records.  Elapsed time: 00:01:38s.  Time for last 1,000,000:    3s.  Last read position: chr2:11,576,972
INFO    2017-10-06 15:47:17     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:22     MarkDuplicates  Read    27,000,000 records.  Elapsed time: 00:01:43s.  Time for last 1,000,000:    4s.  Last read position: chr2:75,878,767
INFO    2017-10-06 15:47:22     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:25     MarkDuplicates  Read    28,000,000 records.  Elapsed time: 00:01:46s.  Time for last 1,000,000:    3s.  Last read position: chr2:149,382,978
INFO    2017-10-06 15:47:25     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:28     MarkDuplicates  Read    29,000,000 records.  Elapsed time: 00:01:49s.  Time for last 1,000,000:    3s.  Last read position: chr2:212,937,960
INFO    2017-10-06 15:47:28     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:33     MarkDuplicates  Read    30,000,000 records.  Elapsed time: 00:01:54s.  Time for last 1,000,000:    4s.  Last read position: chr11:32,426,483
INFO    2017-10-06 15:47:33     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:36     MarkDuplicates  Read    31,000,000 records.  Elapsed time: 00:01:57s.  Time for last 1,000,000:    3s.  Last read position: chr11:104,110,781
INFO    2017-10-06 15:47:36     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:39     MarkDuplicates  Read    32,000,000 records.  Elapsed time: 00:02:01s.  Time for last 1,000,000:    3s.  Last read position: chr17:41,722,964
INFO    2017-10-06 15:47:39     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:43     MarkDuplicates  Read    33,000,000 records.  Elapsed time: 00:02:04s.  Time for last 1,000,000:    3s.  Last read position: chr1:36,095,979
INFO    2017-10-06 15:47:43     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:49     MarkDuplicates  Read    34,000,000 records.  Elapsed time: 00:02:10s.  Time for last 1,000,000:    5s.  Last read position: chr1:101,033,067
INFO    2017-10-06 15:47:49     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:52     MarkDuplicates  Read    35,000,000 records.  Elapsed time: 00:02:13s.  Time for last 1,000,000:    3s.  Last read position: chr1:188,244,158
INFO    2017-10-06 15:47:52     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:55     MarkDuplicates  Read    36,000,000 records.  Elapsed time: 00:02:17s.  Time for last 1,000,000:    3s.  Last read position: chr7:6,484,031
INFO    2017-10-06 15:47:55     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:47:59     MarkDuplicates  Read    37,000,000 records.  Elapsed time: 00:02:20s.  Time for last 1,000,000:    3s.  Last read position: chr7:79,099,764
INFO    2017-10-06 15:47:59     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:48:03     MarkDuplicates  Read    38,000,000 records.  Elapsed time: 00:02:24s.  Time for last 1,000,000:    4s.  Last read position: chr7:145,378,246
INFO    2017-10-06 15:48:03     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:48:07     MarkDuplicates  Read    39,000,000 records.  Elapsed time: 00:02:28s.  Time for last 1,000,000:    3s.  Last read position: chr20:22,620,566
INFO    2017-10-06 15:48:07     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:48:11     MarkDuplicates  Read    40,000,000 records.  Elapsed time: 00:02:32s.  Time for last 1,000,000:    4s.  Last read position: chr5:31,385,718
INFO    2017-10-06 15:48:11     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:48:14     MarkDuplicates  Read    41,000,000 records.  Elapsed time: 00:02:35s.  Time for last 1,000,000:    3s.  Last read position: chr5:100,277,652
INFO    2017-10-06 15:48:14     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:48:19     MarkDuplicates  Read    42,000,000 records.  Elapsed time: 00:02:40s.  Time for last 1,000,000:    4s.  Last read position: chr5:164,625,694
INFO    2017-10-06 15:48:19     MarkDuplicates  Tracking 0 as yet unmatched pairs. 0 records in RAM.
INFO    2017-10-06 15:48:20     MarkDuplicates  Read 42230768 records. 0 pairs never matched.
INFO    2017-10-06 15:48:20     MarkDuplicates  After buildSortedReadEndLists freeMemory: 1034741504; totalMemory: 1059586048; maxMemory: 1059586048
INFO    2017-10-06 15:48:20     MarkDuplicates  Will retain up to 33112064 duplicate indices before spilling to disk.
INFO    2017-10-06 15:48:20     MarkDuplicates  Traversing read pair information and detecting duplicates.
INFO    2017-10-06 15:48:20     MarkDuplicates  Traversing fragment information and detecting duplicates.
INFO    2017-10-06 15:48:30     MarkDuplicates  Sorting list of duplicate records.
INFO    2017-10-06 15:48:31     MarkDuplicates  After generateDuplicateIndexes freeMemory: 783985216; totalMemory: 1058537472; maxMemory: 1058537472
INFO    2017-10-06 15:48:31     MarkDuplicates  Marking 3425825 records as duplicates.
INFO    2017-10-06 15:48:31     MarkDuplicates  Found 0 optical duplicate clusters.
INFO    2017-10-06 15:48:31     MarkDuplicates  Reads are assumed to be ordered by: coordinate
INFO    2017-10-06 15:49:26     MarkDuplicates  Written    10,000,000 records.  Elapsed time: 00:00:55s.  Time for last 10,000,000:   55s.  Last read position: chr19:19,656,741
INFO    2017-10-06 15:50:25     MarkDuplicates  Written    20,000,000 records.  Elapsed time: 00:01:53s.  Time for last 10,000,000:   58s.  Last read position: chr9:29,690,331
INFO    2017-10-06 15:51:24     MarkDuplicates  Written    30,000,000 records.  Elapsed time: 00:02:53s.  Time for last 10,000,000:   59s.  Last read position: chr1:7,582,021
INFO    2017-10-06 15:52:17     MarkDuplicates  Before output close freeMemory: 1062686880; totalMemory: 1073217536; maxMemory: 1073217536
INFO    2017-10-06 15:52:17     MarkDuplicates  After output close freeMemory: 1062675664; totalMemory: 1073217536; maxMemory: 1073217536
[Fri Oct 06 15:52:17 CDT 2017] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 6.65 minutes.
Runtime.totalMemory()=1073217536

968K    /tmp/3512570.mgt2.cluster/rcherniatchik/snappy-1.0.33-libsnappyjava.so
972K    /tmp/3512570.mgt2.cluster/rcherniatchik
972K    total

Disk quotas for user rcherniatchik (uid 10549):
     Filesystem  blocks   quota   limit   grace   files   quota   limit   grace
      /dev/sda3 1225168  20000000 25000000              41       0  800000
   mgt2-ib:/act     540  20000000 25000000              57       0  800000
--------------------------------------------------------------
Begin PBS Epilogue Fri Oct  6 15:52:22 CDT 2017 1507323142
Job ID:                 3512570.mgt2.cluster
Username:               rcherniatchik
Group:                  martyomov
Job Name:               unique_GSM1003475_bams_GSM1003475_hg19
Limits:                 neednodes=1:ppn=4,nodes=1:ppn=4,vmem=32gb,walltime=24:00:00
Resources:              cput=00:07:56,energy_used=0,mem=1250204kb,vmem=5029028kb,walltime=00:06:41
Nodes:                  gpu14
End PBS Epilogue Fri Oct  6 15:52:22 CDT 2017 1507323142
--------------------------------------------------------------
```

**Comment by iromeo** [comment]
Fixed in picard 2.10.7.

**Comment by iromeo** [comment]
Updated picard works ok
